deps/curl-8.6.0/lib/curl_addrinfo.c

```diff
@@ -47,6 +47,10 @@
 #  include <inet.h>
 #endif
 
+#if defined(USE_THREADS_POSIX) && defined(HAVE_PTHREAD_H)
```

Comment:
this if condition is repeated many times - can we create one define for it?

Reply:
Such condition is used in many places in original curl code like here
https://github.com/snowflakedb/libsnowflakeclient/blob/master/deps/curl-8.6.0/lib/easy_lock.h#L96
so I think better to keep it unchanged.